homework1/app.js

```diff
@@ -0,0 +1,650 @@
+        var FoodItem_price = document.getElementById('FoodItem_price').value;
+        var to_push = { name: FoodItem_name, image: FoodItem_image, description: FoodItem_description, price: FoodItem_price };
+
+        if (window.confirm("Do you Really want to add this Food Item?") == true) {
+
+            // check to make sure submitted food is actually new
+            var newFoodItem = true;
+            for (var i = 0; i < storeFoods.length; i++) {
+                if (to_push.name === storeFoods[i].name) {
+                    newFoodItem = false;
+                    break;
+                }
+            }
+            if (newFoodItem) {
+                storeFoods.push(to_push);
+                this.store.foods = storeFoods;
+            }
+        }
+    }
+}
+class Menu {
+    constructor(root, store, cart) {
+        this.root = root;
+        this.store = store;
+        this.cart = cart;
+        this.init();
+    }
+
+    init() {
+        this.render();
+    }
+
+    render() {
+
+        let tbody = this.root.querySelector('tbody');
+
+        for (var i = 0; i < this.store.foods.length; i++) {
+
+            var FoodItem_name = this.store.foods[i].name;
+            var FoodItem_image = this.store.foods[i].image;
+            var FoodItem_description = this.store.foods[i].description;
+            var FoodItem_price = this.store.foods[i].price;
+            tbody.innerHTML +=
+                `<tr >
+                    <td>${FoodItem_name}</td>
+                    <td><img  src=${FoodItem_image} ></td>
+                    <td>${FoodItem_description}</td>
+                    <td>$${FoodItem_price}</td>
+                     <td>
+							<button  class="checkout-button" data-name="${FoodItem_name}" data-image="${FoodItem_image}"  data-description="${FoodItem_description}" data-price="${FoodItem_price}"  data-quantity="1" >
+								Add to the Cart
+							</button>
+                        </td>
+                 </tr>`;
+        }
+
+        let checkoutButtons = document.querySelectorAll('.checkout-button');
+        if (checkoutButtons && checkoutButtons.length) {
+            for (var i = 0; i < checkoutButtons.length; i++) {
+                new CheckoutButton(checkoutButtons[i], this.store, this.cart);
+            }
+        }
+    }
+}
+class OrderStatusControlTable {
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+        this.init();
+    }
+
+    init() {
+        this.render();
+    }
+
+    destroy() {
+        // removes EventListeners for updateOrderStatusControlTable buttons
+    }
+
+    render() {
+        let tbody = this.root.querySelector('tbody');
+        tbody.innerHTML = ``;
+        if (this.store.queue === null) {
+            this.store.queue = [];
+        }
+        for (var i = 0; i < this.store.queue.length; i++) {
+           
+            var status_name = Item + " Status";
```

Comment:
Syntax error and therefore everything afterward is not executable (not testable).